tests/Sniffs/Files/TypeNameMatchesFileNameSniffTest.php

```diff
@@ -84,4 +86,14 @@ public function testRootNamespacesNormalization(): void
 		self::assertNoSniffErrorInFile(self::checkFile(__DIR__ . '/data/rootNamespace2/Xxx/Boo.php', $sniffProperties2));
 	}
 
+	public function testWithProvidedBasePathAndNestedSameDirName(): void
+	{
+		$filePath = str_replace('/', DIRECTORY_SEPARATOR, __DIR__ . '/data/data/Foo/Bar.php');
```

Comment:
@rmikalkenas This is wrong. You need to normalize it in `TypeNameMatchesFileNameSniff.php`, line 93.

Reply:
are you sure? Because `$phpcsFile->getFilename();` is already normalized by `squizlabs/php_codesniffer`.
I think it should be normalized at test, because filepath Im writing is with forwardslash and windows uses backslack

Reply:
Yes, I'm sure. You cannot rely on PHPCS behaviour. It should be normalized just before the check.